README.md

```diff
@@ -4,569 +4,1111 @@
+
+```ruby
+class PostImageUploader < Saviour::BaseUploader
+  store_dir { "uploads/posts/#{model.id}/#{attached_as}" }
+
+  process_with_file do |file, filename|
+    `convert -thumbnail 100x100^ #{Shellwords.escape(file.path)}`
+    
+    [file, filename] 
+  end
+end
 ```
 
-The `local_prefix` option is mandatory, and defines the base prefix under which the storage will store files in the
-machine. You need to configure this accordingly to your use case and deployment strategies, for example, for rails
-and capistrano with default settings you'll need to set it to `Rails.root.join("public/system")`.
+*Note that when escaping to shell you need to check for safety in case there's an injection in the filename.*
+
+You can modify directly the contents of the given file in the filesystem, or you could also delete the given file and
+return a new one. If you return a different file instance, you're expected to clean up the one that was given to you.
+
+You can mix `process` with `process_with_file` but you should try to avoid it, as it will be a performance penalty
+having to convert between formats.
+
+Also, even if there's just one `process`, the whole contents of the file will be loaded into memory. Avoid its usage 
+if you're conservative about memory usage, or take care of restricting the allowed file size you can work with on
+any file upload you accept across your application.
 
-The `public_url_prefix` is optional and should represent the public endpoint from which you'll serve the assets.
-Same as before, you'll need to configure this accordingly to your deployment specifics.
-You can also assign a Proc instead of a String to dynamically manage this (for multiple asset hosts for example).
 
-This storage will take care of removing empty folders after removing files.
+##### halt_process
 
+`halt_process` is a method you can call from inside a processor in order to abort the processing and storing
+of the current file. You can use this to conditionally store a file or not based on runtime decisions.
 
-### S3Storage
+For example, you may be storing media files that can be audio, video or images, and you want to generate a
+thumbnail for videos and images but not for audio files.
 
-An storage implementation using `Fog::AWS` to talk with Amazon S3. Example:
+```ruby
+class ThumbImageUploader < Saviour::BaseUploader
+  store_dir { "uploads/thumbs/#{model.id}/#{attached_as}" }
+
+  process_with_file do |file, filename|
+    halt_process unless can_generate_thumb?(file)
+    `convert -thumbnail 100x100^ #{Shellwords.escape(file.path)}`
 
+    [file, filename]
+  end
+
+  def can_generate_thumb?(file)
+    # Some mime type checking
+  end
+end
 ```
-Saviour::Config.storage = Saviour::S3Storage.new(
-  bucket: "my-bucket-name",
-  aws_access_key_id: "stub",
-  aws_secret_access_key: "stub"
-)
+
+
+### Versions
+
+Saviour has no concept of "file versions" as found in other file attachment libraries. The only facilities provided
```

Comment:
not following that intro, what are the differences then? You could have even more flexibility why putting it as a downside? :)